python/benchmark/test_gen_data.py

```diff
@@ -416,9 +419,10 @@ def test_make_sparse_regression(
                 assert dense_count >= chunk_size * num_partitions * int(
                     (row_num // num_partitions) * col_density - 1
                 ) and dense_count <= chunk_size * num_partitions * int(
-                    (row_num // num_partitions) * col_density + 1
+                    (row_num // num_partitions + 1) * col_density + 1
```

Comment:
Is this the actual guarantee on dense_count you are aiming for?

Reply:
Yes, so for the number of rows in each partition it's either (row_num // numpartitions) or (row_num // numpartitions + 1). Then the upper bound should have a plus for corner case (Say that row_num mod num_partitions = num_partition - 1).